### PR TITLE
users: add documentation for the users role

### PR DESCRIPTION
**Pull request by andrewschoen**
Documentation for the users role.  I think we should start putting README files in each of the roles like this. Maybe eventually we can create a sphinx project that will collect all those and build docs that we could host somewhere.

Grammar nazi's welcome :)

### CODE REVIEW

**Comment by ktdreyer** [comment]
nazi's -> nazis

**Comment by andrewschoen** [comment]
Lol, thank you.

File: roles/users/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        key: <ssh_key_string>
+      - name: user2
+        key: <ssh_key_url>
+
+    managed_admin_users:
+      - name: admin
+        key: <ssh_key_string>
+
+The key for a user can either be an SSH public key as a string or url. Users are created
+with 'bin/bash' as their default shell.
```
Comment (alfredodeza) [inline roles/users/README.rst]: /bin/bash

File: roles/users/README.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
+Users
+=====
+
+This role is used to manage user accounts on a node.  It is capable of setting up users
+with and without sudo access, but does not correctly have the ability to remove users.
```
Comment (zmc) [inline roles/users/README.rst]: "does not correctly have" ? 

Do you mean 'currently'?

File: roles/users/README.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+For example, this would create and update keys for all users defined for $NODE. First, be
+sure to define either ``managed_users`` or ``managed_admin_users`` in your inventory::
+
+    $ ansible-playbook users.yml --limit="$NODE"
+
+Variables
++++++++++
+
+Available variables are listed below, along with default values (see `defaults/main.yml`):
```
Comment (alfredodeza) [inline roles/users/README.rst]: I think you want ``defaults/main.yml`` here  (with double backticks)

**Comment by zmc** [comment]
Let's make sure @dmick sees this!

File: roles/users/README.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+A list of hashes that define users that will be created **without** sudo access::
+
+    managed_users: []
+
+A list of hashes taht define users that will be created **with** sudo access::
```
Comment (alfredodeza) [inline roles/users/README.rst]: s/taht/that/g

**Comment by andrewschoen** [comment]
@zmc agreed, I was hoping @dmick would have some notes / feedback on this.

File: roles/users/README.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+Usage
++++++
+
+This role is required as a dependency for the ``common`` role so it's already in use for most
+all groups and playbooks, but if you need to manage users for a specific node or for a
+one-off situation you can use the users.yml playbook.
+
+For example, this would create and update keys for all users defined for $NODE. First, be
+sure to define either ``managed_users`` or ``managed_admin_users`` in your inventory::
```
Comment (dmick) [inline roles/users/README.rst]: s/inventory/inventory; then/

File: roles/users/README.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+pubkeys
+    Perform only authorized keys tasks, users will not be created but all
+    SSH keys will be updated for both ``managed_users`` and ``managed_admin_users``.
+
+TODO
+++++
+
+- Allow management of the UID for each user
+
+- Allow managment of the shell for each user
```
Comment (dmick) [inline roles/users/README.rst]: s/managment/management/

File: roles/users/README.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
+Users
+=====
+
+This role is used to manage user accounts on a node.  It is capable of setting up users
+with and without sudo access, but does not currently have the ability to remove users.
+In either your group_vars or host_vars files you can define two variables for this role
```
Comment (dmick) [inline roles/users/README.rst]: s/can/must/

File: roles/users/README.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+to use: ``managed_users`` and ``managed_admin_users``.  The ``managed_users`` variable
+will create users without sudo access while users in the ``managed_admin_users`` list
+will be granted sudo access. Sudo access is granted by adding the ``managed_admin_users`` to
+the group ``sudo`` which should be created beforehand.
+
+For example, in inventory/group_vars/webservers.yml you might have a list of users like this::
```
Comment (dmick) [inline roles/users/README.rst]: Seems better ot move this down to the vars section; then you don't have to repeat the bit about it being a string or a url, either

File: roles/users/README.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
+Users
+=====
+
+This role is used to manage user accounts on a node.  It is capable of setting up users
```
Comment (dmick) [inline roles/users/README.rst]: s/setting up/adding/

File: roles/users/README.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+with and without sudo access, but does not currently have the ability to remove users.
+In either your group_vars or host_vars files you can define two variables for this role
+to use: ``managed_users`` and ``managed_admin_users``.  The ``managed_users`` variable
+will create users without sudo access while users in the ``managed_admin_users`` list
+will be granted sudo access. Sudo access is granted by adding the ``managed_admin_users`` to
```
Comment (dmick) [inline roles/users/README.rst]: Maybe say briefly what the ... playbook?... does in total; I assume it's "add the user, create a home dir, add .ssh/authorized_keys, etc.  and maybe note which tasks are accomplished by the "user" tag and which are in the "pubkeys" tag

**Comment by andrewschoen** [comment]
@dmick I've got your review feedback done in https://github.com/ceph/ceph-cm-ansible/commit/4e64ae788630f5d0cf0b052e054a0d6d854140d0

**Comment by andrewschoen** [comment]
Alright, this should be good to go unless anybody else has feedback.

**Pull request merged**